homeassistant/components/kostal_piko/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Kostal Piko sensors."""
+import logging
+
+import kostal
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PikoUpdateCoordinator
+from .const import DOMAIN, SENSOR_TYPES, KostalPikoEntityDescription
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Kostal Piko platform with its sensors."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    device_info = DeviceInfo(
+        configuration_url=entry.data[CONF_HOST],
+        identifiers={(DOMAIN, coordinator.data[kostal.InfoVersions.SERIAL_NUMBER])},
+        manufacturer="Kostal",
+        model=coordinator.data[kostal.SettingsGeneral.INVERTER_MAKE],
+        name=coordinator.data[kostal.SettingsGeneral.INVERTER_NAME],
+        sw_version=coordinator.data[kostal.InfoVersions.VERSION_FW],
+        hw_version=coordinator.data[kostal.InfoVersions.VERSION_HW],
+    )
+
+    async_add_entities(
+        KostalPikoSensor(coordinator, description, device_info)
+        for description in SENSOR_TYPES
+    )
+
+
+class KostalPikoSensor(CoordinatorEntity[PikoUpdateCoordinator], SensorEntity):
+    """A Kostal Piko sensor updated using a DataUpdateCoordinator."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: PikoUpdateCoordinator,
+        description: KostalPikoEntityDescription,
+        deviceInfo: DeviceInfo,
+    ) -> None:
+        """Create a new KostalPikoSensor entity for inverter data."""
+        super().__init__(coordinator)
+        self.dxs_id = int(description.key)
+        self.formatter = description.formatter
+        self._attr_device_info = deviceInfo
```

Comment:
```suggestion
        self._attr_device_info = DeviceInfo(
            configuration_url=entry.data[CONF_HOST],
            identifiers={(DOMAIN, coordinator.data[kostal.InfoVersions.SERIAL_NUMBER])},
            manufacturer="Kostal",
            model=coordinator.data[kostal.SettingsGeneral.INVERTER_MAKE],
            name=coordinator.data[kostal.SettingsGeneral.INVERTER_NAME],
            sw_version=coordinator.data[kostal.InfoVersions.VERSION_FW],
            hw_version=coordinator.data[kostal.InfoVersions.VERSION_HW],
        )
```

Makes the code more readable